File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/TaskBaselines/createtaskbaseline.rb
module Asposetasksjava
  module CreateTaskBaseline
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')
		
    	# Set baseline for the entire project
    	project.setBaseline(Rjb::import('com.aspose.tasks.BaselineType').Baseline)

    	puts "Set baseline for the project."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTaskBaselines/CreateTaskBaseline.php
<?php
namespace Aspose\Tasks\WorkingWithTaskBaselines;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\BaselineType as BaselineType;
class CreateTaskBaseline{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

    	# Set baseline for the entire project
        $baselineType=new BaselineType();
    	$project->setBaseline($baselineType->Baseline);

    	print "Set baseline for the project.".PHP_EOL;
    }

}
?>